### PR TITLE
Add Korean translation for glossary.md

### PR DESCRIPTION
**Pull request by JoosunH**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by JoosunH** [comment]
Hi @jungnerd , I've finished translating glossary.md into Korean.
Following the i18n guidelines — would appreciate your review when you have time 😊
Thank you!

**Comment by Rocketknight1** [comment]
cc @stevhliu

File: docs/source/ko/glossary.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+### 포지션 ID (position IDs)
+
+RNN 모델과 달리 트랜스포머는 각 토큰의 위치 정보를 내부적으로 가지고 있지 않습니다. 따라서 모델은 `position_ids`를 사용하여 각 토큰이 시퀀스 내에서 어느 위치에 있는지를 인식합니다. 이 값은 선택적인 파라미터입니다. 모델에 `position_ids`를 전달하지 않으면, 절대 위치 임베딩 방식으로 자동 생성됩니다. 절대 위치 임베딩은 `[0, config.max_position_embeddings - 1]` 범위 내에서 선택됩니다. 일부 모델은 사인파 형태의 위치 임베딩(sinusoidal position embeddings) 또는 상대 위치 임베딩(relative position embeddings)과 같은 다른 유형의 위치 임베딩을 사용하기도 합니다.
+
+### 전처리 (preprocessing)
+
+머신러닝 모델이 쉽게 처리할 수 있도록 가공되지 않은 데이터를 정제하는 작업입니다. 예를 들어, 텍스트는 일반적으로 토크나이징(tokenization) 과정을 거칩니다. 다른 입력 유형에 대한 전처리 방식이 궁금하다면 [Preprocess](https://huggingface.co/docs/transformers/preprocessing) 튜토리얼을 참고해 보세요.
+
+### **"사전 학습된 모델 (pretrained model)"**
```
Comment (stevhliu) [inline docs/source/ko/glossary.md]:
```suggestion
### 사전 학습된 모델 (pretrained model)
```

**Review by stevhliu — APPROVED** [review]
LGTM, thanks for this! We can merge once @jungnerd approves 🙂

File: docs/source/ko/glossary.md [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+```
+
+이것은 PyTorch나 TensorFlow의 텐서로 변환될 수 있습니다. 어텐션 마스크는 모델이 패딩 된 인덱스를 참조하지 않도록 해당 위치를 나타내는 이진 텐서입니다. [`BertTokenizer`]의 경우, `1`은 어텐션이 필요한 값을 나타내고, `0`은 패딩 된 값을 나타냅니다. 이 어텐션 마스크는 토크나이저가 반환되는 딕셔너리의 "attention_mask" 키 아래에 포함되어 있습니다:
+
+```python
+>>> padded_sequences["attention_mask"]
+[[1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]
+```
+
+### autoencoding models
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 오토인코딩 모델 (autoencoding models)
```

File: docs/source/ko/glossary.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+
+[인코더 모델](#encoder-models)과 [마스킹된 언어 모델링](#masked-language-modeling-mlm)을 참고하세요.
+
+### autoregressive models
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 자기회귀 모델 (autoregressive models)
```

File: docs/source/ko/glossary.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+
+[인과적 언어 모델링](#causal-language-modeling)과 [디코더 모델](#decoder-models)을 참고하세요.
+
+## B
+
+### backbone
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 백본 (backbone)
```

File: docs/source/ko/glossary.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+백본(backbone)은 원시(hidden) 은닉 상태(hidden state) 또는 특징(feature)을 출력하는 네트워크(임베딩과 레이어)입니다. 일반적으로 이 백본은 해당 특징을 입력으로 받아 예측을 수행하는 [헤드](#head)와 연결됩니다. 예를 들어, [`ViTModel`]은 특정 헤드가 없는 백본입니다. 다른 모델들도[`VitModel`]을 백본으로 사용할 수 있으며, [DPT](model_doc/dpt)등이 그 예시입니다.
+
+## C
+
+### 인과적 언어 모델링 (causal language modeling)
+
+모델이 텍스트를 순서대로 읽으며 다음 단어를 예측해야 하는 사전 학습(pretraining)과제입니다. 일반적으로 문장을 전체로 읽되, 모델 내부에서 특징 시점 이후의 토큰을 마스킹(masking)하여 다음 단어를 예측하게 됩니다.
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
모델이 텍스트를 순서대로 읽으며 다음 단어를 예측해야 하는 사전 학습(pretraining) 작업입니다. 일반적으로 문장을 전체로 읽되, 모델 내부에서 특징 시점 이후의 토큰을 마스킹(masking)하여 다음 단어를 예측하게 됩니다.
```
task를 '과제'라고 번역하는 것보다 '작업' 또는 '태스크'로 번역하는 것이 자연스러운 것 같아 수정해보았습니다.

File: docs/source/ko/glossary.md [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+
+### 채널 (channel)
+
+컬러 이미지는 빨간색(R), 초록색(G), 파란색(B)의 세 채널 값을 조합하여 구성되며, 흑백 이미지는 단일 채널만을 가집니다. 🤗 Transformers에서는 이미지 텐서의 채널이 첫 번째 또는 마지막 차원에 위치할 수 있습니다:[`n_channels`, `height`, `width`] 또는 [`height`, `width`, `n_channels`]와 같은 형식입니다.
+
+### 연결 시간분류(connectionist temporal classification)
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 연결 시간분류(connectionist temporal classification, CTC) 
```
원문에서는 약어(CTC)를 표현해두어서, 이 문서에서도 함께 표기했습니다!

**Review by jungnerd — COMMENTED** [review]
안녕하세요! 좋은 번역 정말 감사합니다 🤗
리뷰를 하며 문서를 읽어보니 굉장히 유익하네요!

현재 C 파트까지 리뷰 완료했고, 곧 이어서 리뷰하겠습니다.

I’ve reviewed up to the C section, and I will continue the rest of the docs shortly.

File: docs/source/ko/glossary.md [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+
+## D
+
+### 데이터 병렬화 (DataParallel)
+
+여러 개의 GPU에서 훈련을 수행할 때 사용하는 병렬화 기법으로, 동일한 모델 구성이 여러 번 복제되며 각 인스턴스는 서로 다른 데이터 조각을 받습니다. 모든 인스턴스는 병렬로 처리를 수행하며, 각 훈련 단계가 끝난 후 결과를 동기화합니다.
+
+DataParallel 방식에 대해 더 알아보려면 [여기](perf_train_gpu_many#dataparallel-vs-distributeddataparallel)를 참고하세요.
+
+### decoder input IDs
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 디코더 입력 ID (decoder input IDs)
```
아래에서 `input IDs` 를 `입력 ID`로 번역해두셨길래 여기도 수정했습니다.

File: docs/source/ko/glossary.md [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+
+이 입력은 인코더-디코더 모델에 특화된 것으로, 디코더에 전달될 input ID 들을 포함합니다. 이러한 입력은 번역이나 요약과 같은 시퀀스-투-시퀀스(sequence-to-sequence) 작업에 사용되며, 일반적으로 모델마다 고유한 방식으로 구성됩니다.
+
+대부분의 인코더-디코더 모델(BART, T5 등)은 `labels`로부터 자동으로 `decoder_input_ids`를 생성합니다. 이러한 모델에서는 학습 시 `labels`를 전달하는 것이 일반적으로 권장됩니다.
+
+시퀀스-투-시퀀스 학습에서 각 모델이 이러한 input ID를 어떻게 처리하는지는 모델 문서를 참고하시기를 바랍니다.
+
+### 디코더 모델 (decoder models)
+
+오토리그레시브 모델(Autoregressive models)이라고도 불리는 디코더 모델은 인과 언어 모델링(causal language modeling)이라 불리는 사전 학습 작업을 수행합니다. 이 작업에서는 모델이 텍스트를 순서대로 읽고 다음 단어를 예측해야 합니다. 일반적으로 문장의 전체를 읽되, 특정 시점 이후의 토큰은 마스크로 가려 예측하게 합니다.
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
자기회귀 모델(Autoregressive models)이라고도 불리는 디코더 모델은 인과 언어 모델링(causal language modeling)이라 불리는 사전 학습 작업을 수행합니다. 이 작업에서는 모델이 텍스트를 순서대로 읽고 다음 단어를 예측해야 합니다. 일반적으로 문장의 전체를 읽되, 특정 시점 이후의 토큰은 마스크로 가려 예측하게 합니다.
```
Autoregressive model을 `자기회귀 모델`로 많이 번역하는 것 같아서 수정했습니다!

File: docs/source/ko/glossary.md [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+
+<Youtube id="d_ixlCubqQw"/>
+
+### 딥러닝 (deep learning)
+
+여러층의 신경망 (neural network)을 사용하는 머신러닝 알고리즘입니다.
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
여러 층의 신경망(neural network)을 사용하는 머신러닝 알고리즘입니다.
```
띄어쓰기 수정했습니다.

File: docs/source/ko/glossary.md [hunk 10]
```diff
@@ -0,0 +1,454 @@
...
+
+<Youtube id="H39Z_720T5s"/>
+
+## F
+
+### 특징 추출 (feature extraction)
+
+머신러닝 알고리즘이 더 효과적으로 학습할 수 있도록, 원시 데이터를 선택하고 변환하여 더 유용한 특징(feature) 집합으로 만드는 과정입니다. 예를 들어, 원시 텍스트를 워드 임베딩으로 변환하거나 이미지나 비디오 데이터에서 윤곽선이나 형태와 같은 중요한 특징을 추출하는 것이 있습니다.
+
+### 피드포워드 청킹 (feed forward chunking)
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 피드 포워드 청킹 (feed forward chunking)
```

File: docs/source/ko/glossary.md [hunk 11]
```diff
@@ -0,0 +1,454 @@
...
+
+트랜스포머의 각 residual attention Block에서는 self-Attention Layer 다음에 보통 두 개의 Feed Forward Layer가 이어집니다. 이 Feed Forward Layers의 중간 임베딩 크기는 종종 모델의 히든 사이즈(hidden size)보다 큽니다(예:
+`google-bert/bert-base-uncased` 모델의 경우).
+
+입력 크기가 `[batch_size, sequence_length]`일 경우, 중간 Feed Forward 임베딩
+`[batch_size, sequence_length, config.intermediate_size]`을 저장하는 데 필요한 메모리는 전체 메모리 사용량의 큰 부분을 차지할 수 있습니다.
+[Reformer: The Efficient Transformer](https://arxiv.org/abs/2001.04451) 논문의 저자들은 이 연산이 `sequence_length` 차원에 대해 독립적이기 때문에,토큰마다 Feed Forward Layer의 출력 임베딩을 각 토큰별로 `[batch_size, config.hidden_size]`을 개별적으로 계산한 뒤, 이를 이어 붙여 `[batch_size, sequence_length, config.hidden_size]` 형태로 만들 수 있습니다.`n = sequence_length`. 이 방식은 계산 시간은 늘어나지만, 메모리 사용량은 줄어들게 됩니다.
+
+[`apply_chunking_to_forward`] 함수를 사용하는 모델의 경우, `chunk_size`는 병렬로 계산되는 출력 임베딩의 개수를 정의하며, 이는 메모리 사용량과 계산 시간 간의 트레이드오프를 결정합니다.
+`chunk_size`가 0으로 설정되면, Feed Forward 청킹은 수행되지 않습니다.
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
`chunk_size`가 0으로 설정되면, 피드 포워드 청킹(Feed Forward Chunking)은 수행되지 않습니다.
```

File: docs/source/ko/glossary.md [hunk 12]
```diff
@@ -0,0 +1,454 @@
...
+
+행렬의 특정 차원에서 최댓값이나 평균값을 취하여 더 작은 행렬로 줄이는 연산입니다. 풀링 계층은 주로 합성곱 계층 사이에 위치하여 특징 표현을 다운샘플링 하는 데 사용됩니다.
+
+### 포지션 ID (position IDs)
+
+RNN 모델과 달리 트랜스포머는 각 토큰의 위치 정보를 내부적으로 가지고 있지 않습니다. 따라서 모델은 `position_ids`를 사용하여 각 토큰이 시퀀스 내에서 어느 위치에 있는지를 인식합니다. 이 값은 선택적인 파라미터입니다. 모델에 `position_ids`를 전달하지 않으면, 절대 위치 임베딩 방식으로 자동 생성됩니다. 절대 위치 임베딩은 `[0, config.max_position_embeddings - 1]` 범위 내에서 선택됩니다. 일부 모델은 사인파 형태의 위치 임베딩(sinusoidal position embeddings) 또는 상대 위치 임베딩(relative position embeddings)과 같은 다른 유형의 위치 임베딩을 사용하기도 합니다.
+
+### 전처리 (preprocessing)
+
+머신러닝 모델이 쉽게 처리할 수 있도록 가공되지 않은 데이터를 정제하는 작업입니다. 예를 들어, 텍스트는 일반적으로 토크나이징(tokenization) 과정을 거칩니다. 다른 입력 유형에 대한 전처리 방식이 궁금하다면 [Preprocess](https://huggingface.co/docs/transformers/preprocessing) 튜토리얼을 참고해 보세요.
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
머신러닝 모델이 쉽게 처리할 수 있도록 가공되지 않은 데이터를 정제하는 작업입니다. 예를 들어, 텍스트는 일반적으로 토큰화(tokenization) 과정을 거칩니다. 다른 입력 유형에 대한 전처리 방식이 궁금하다면 [Preprocess](https://huggingface.co/docs/transformers/preprocessing) 튜토리얼을 참고해 보세요.
```
토크나이징 -> 토큰화 로 수정해보았습니다.

File: docs/source/ko/glossary.md [hunk 13]
```diff
@@ -0,0 +1,454 @@
...
+
+소량의 라벨이 달린 데이터와 대량의 라벨이 없는 데이터를 함께 사용하여 모델의 정확도를 높이는 머신러닝 훈련 기법의 넓은 범주입니다. 이는 [지도 학습](#supervised-learning)이나 [비지도 학습](#unsupervised-learning)과는 다른 방식입니다.
+
+준지도 학습 기법의 예로는 "자기 학습(self-training)"이 있습니다. 이 방식은 먼저 라벨이 있는 데이터로 모델을 학습시키고, 그 모델을 사용해 라벨이 없는 데이터에 대한 예측을 수행합니다. 모델이 가장 높은 확신을 가지고 예측한 라벨이 없는 데이터 일부를 라벨이 있는 데이터로 추가하고, 이를 통해 모델을 다시 학습시킵니다.
+
+### 시퀀스 투 시퀀스 (seq2seq)
+
+입력으로부터 새로운 시퀀스를 생성하는 모델입니다. 예를 들어 번역 모델이나 요약 모델이 이에 해당하며, 대표적인 예로는 [Bart](model_doc/bart)나[T5](model_doc/t5) 모델이 있습니다.
+
+### 샤디드 DDP (Sharded DDP)
```
Comment (wo0jun) [inline docs/source/ko/glossary.md]:
```suggestion
### 분할 DDP (Sharded DDP)
```
[다른 transformers 문서](https://huggingface.co/docs/transformers/ko/fsdp)에서 `Fully Sharded Data Parellel`을 `완전 데이터 분할 처리`로 번역한 것을 참고해서 수정해보았습니다.

**Review by jungnerd — COMMENTED** [review]
나머지 부분도 리뷰해보았습니다! 
헤딩 중 번역이 안된 부분에는 번역을 추가하였고, 일부 단어에 대해서는 다른 번역을 제안드렸습니다. 😊
좋은 번역 정말 감사합니다!

**Review by jungnerd — COMMENTED** [review]
나머지 부분도 리뷰해보았습니다! 
헤딩 중 번역이 안된 부분에는 번역을 추가하였고, 일부 단어에 대해서는 다른 번역을 제안드렸습니다. 😊
좋은 번역 정말 감사합니다!

**Comment by JoosunH** [comment]
감사합니다! 모든 제안사항 반영하였습니다! 확인 부탁드립니다!

**Comment by wo0jun** [comment]
Hi @stevhliu 👋
The Korean translations look great to me. I believe this PR is ready to be merged!

**Comment by stevhliu** [comment]
Thank you both for iterating and contributing! 🤗
